lib/devise-security/validators/complexity_validator.rb

```diff
@@ -0,0 +1,23 @@
+module DeviseSecurity
+  # String complexity validator
+  # Validates complexity of e.g. passwords
+  # Options:
+  # - digit: minimum number of digits in the validated string
+  # - lower: minimum number of lower-case letters in the validated string
+  # - punct: minimum number of punctuation characters or symbols in the validated string
```

Comment:
Agree on the character classes, that makes sense. 

Not sure what you mean about 'sets of repeated characters or sequences', though? Each regex is for a single character so I'm not sure how repeated characters or sequences in the input would cause a problem? 

Another approach would be simply to iterate over each character in the string in turn, bumping the count for any character set it meets (using `unpack('U')` or similar) and do away with regexen altogether. Thoughts?

Reply:
Some recommendations for password complexity suggest that you should avoid passwords with long sets of repeated characters (i.e. `aaaaaa`) and sequential characters (`abcdefg` or `1234`).  This is outside the scope of this specific PR as the original code didn't do that either.

Reply:
Using `unpack('U')` is an interesting approach.

Reply:
Oh I _see_. When you said 'protect against', I thought you were talking about protecting against ReDOS attacks (which, incidentally, use repetition, etc).

Reply:
Just about to head home for the weekend, but I should be able to update this PR on Monday (Australia time - so that's Sunday eve from your perspective :)

Reply:
And yes, agreed repeated characters would be useful to protect against (but out of scope for this PR, which is just to use a safer method than asking developers to write a regex). Another common request is avoidance of common dictionary words. 

In fact, I don't really think much of the usual complexity rules; I agree with Jeff Atwood that [password rules are bullsh**](https://blog.codinghorror.com/password-rules-are-bullshit/). (And indeed [Bill Burr](https://www.wsj.com/articles/the-man-who-wrote-those-password-rules-has-a-new-tip-n3v-r-m1-d-1502124118).) I'd much rather use a validator like this one: 
https://github.com/cmer/nobspw

However, our clients in their wisdom want the usual kind of complexity rules, based on outdated standards...and they're paying the bills 😂